lib/download.sh

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env bash
+#
+# utils to download and verify binary downloads in shell scripts
+#
+# this expects lib/common.sh to be sourced
+
+wget_and_verify()
+{
+    local url="${1:?url missing}"
+    local sha256="${2:?sha256 missing}"
+    local target="${3:?target missing}"
+    local checksum
+
+    declare -a args=(
+        --no-verbose
+        -O "${target}"
+        "${url}"
+    )
+
+    wget "${args[@]}"
+
+    checksum="$(sha256sum "${target}" | awk '{print $1;}')"
+    if [[ "${checksum}" != "${sha256}" ]]; then
+        echo >&2 "fatal: ${url} binary checksum '${checksum}' differs from expected checksum '${sha256}'"
+        return 1
+    fi
+
+    return 0
+}
+
+download_and_install_krew()
+{
+    local tmp_dir
+    tmp_dir="$(mktemp -d)"
+    pushd "${tmp_dir}" || return 1
+
+    KERNEL_OS="$(uname | tr '[:upper:]' '[:lower:]')"
+    ARCH="$(uname -m | sed -e 's/x86_64/amd64/' -e 's/\(arm\)\(64\)\?.*/\1\2/' -e 's/aarch64$/arm64/')"
+    KREW="krew-${KERNEL_OS}_${ARCH}"
+    # TODO: can't pin latest
+    KREW_URL="https://github.com/kubernetes-sigs/krew/releases/download/${KREW_VERSION}/${KREW}.tar.gz"
+    wget_and_verify "${KREW_URL}" "${KREW_SHA256}" "${KREW}.tar.gz"
+    tar zxvf "${KREW}.tar.gz"
+    rm -f "${KREW}.tar.gz"
+    ./"${KREW}" install krew
+
+    # Add krew to PATH by appending this line to .bashrc
+    krew_path_bashrc="export PATH=${KREW_ROOT:-${HOME}/.krew}/bin:${PATH}"
+    # Add the line if it is not already there
+    grep -qxF "${krew_path_bashrc}" ~/.bashrc || echo "${krew_path_bashrc}" >> ~/.bashrc
+    popd || return 1
+}
+
+download_and_install_minikube()
+{
+    MINIKUBE_URL="https://storage.googleapis.com/minikube/releases/${MINIKUBE_VERSION}/minikube-linux-amd64"
+    MINIKUBE_BINARY="minikube"
+
+    wget_and_verify "${MINIKUBE_URL}" "${MINIKUBE_SHA256}" "${MINIKUBE_BINARY}"
+    chmod +x "${MINIKUBE_BINARY}"
+    sudo mv "${MINIKUBE_BINARY}" /usr/local/bin/
+}
+
+download_and_install_kvm2_driver()
+{
+    DRIVER_URL="https://storage.googleapis.com/minikube/releases/${MINIKUBE_VERSION}/docker-machine-driver-kvm2"
+    DRIVER_BINARY="docker-machine-driver-kvm2"
+
+    wget_and_verify "${DRIVER_URL}" "${MINIKUBE_DRIVER_SHA256}" "${DRIVER_BINARY}"
+    chmod +x "${DRIVER_BINARY}"
+    sudo mv "${DRIVER_BINARY}" /usr/local/bin/
+}
+
+download_and_install_kind()
+{
+    KIND_URL="https://github.com/kubernetes-sigs/kind/releases/download/${KIND_VERSION}/kind-$(uname)-amd64"
+    KIND_BINARY="kind"
+
+    wget_and_verify "${KIND_URL}" "${KIND_SHA256}" "${KIND_BINARY}"
+    chmod +x "${KIND_BINARY}"
+    sudo mv "${KIND_BINARY}" /usr/local/bin/
+}
+
+download_and_install_tilt()
+{
+    # TODO: see if there is actual releases instead of just git sha
```

Comment:
Yes there are releases and it is possible to get the binaries directly for them. I do it like this in ansible: https://github.com/lentzi90/workstation/blob/8c1dc0e209311a8380683e652b46e2f42b132015/roles/workstation/tasks/tilt.yml#L24-L30

Reply:
It was more of a note to myself, and yes, I've got the release variables defined now.